cmd/generate/main.go

```diff
@@ -8,58 +8,107 @@
+			"You should provide it here or in json config.")
 		matchFuncName = flag.String("match-func-name", "", "Name of the generated matchFunc."+
-			" Normally it should start with uppercase to export. (required)")
+			"Normally it should start with uppercase to export. If this is provided, the match-func-name in json config will be ignored.\n"+
+			"You should provide it here or in json config.")
 		matchFnComment = flag.String("match-func-comment", "", "The code comment for generated matchFunc."+
-			" You can add some explanations like which modules included in it. Normally it should start with match-func-name (optional)")
+			"You can add some explanations like which modules included in it. Normally it should start with match-func-name.\n"+
+			"If this is provided, the matchFuncComment in json config will be ignored. (optional)")
 		filterflags       filterFlag
 		directiveOverride override
 	)
 	flag.Var(&filterflags, "filter",
 		"A list of strings specifying the directives to exclude from the output. "+
-			"An example is: -filter directive1 -filter directive2... (optional)")
+			"An example is: -filter directive1 -filter directive2...\n"+
+			"If this is provided, the filter in json config will be igonored. (optional)")
 	flag.Var(&directiveOverride, "override",
 		"A list of strings, used to override the output. "+
 			"It should follow the format:{directive:bitmask00|bitmask01...,bitmask10|bitmask11...}"+"\n"+
 			"An example is -override=log_format:ngxHTTPMainConf|ngxConf2More,ngxStreamMainConf|ngxConf2More"+"\n"+
-			`To use | and , in command line, you may need to enclose your input in quotes, i.e. -override="directive:mask1,mask2,...". (optional)`)
+			`To use | and , in command line, you may need to enclose your input in quotes, i.e. -override="directive:mask1,mask2,...".`+
+			`If this is provided, the override in json config will be ignored. (optional)`)
 
 	flag.Parse()
+	var err error
 
-	if *sourceCodePath == "" {
-		log.Fatal("src-path can't be empty")
+	var config generator.GenerateConfig
+
+	if *configPath != "" {
+		config, err = configFromFile(*configPath)
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
 
-	if *directiveMapName == "" {
-		log.Fatal("directive-map can't be empty")
+	if *directiveMapName != "" {
+		config.DirectiveMapName = *directiveMapName
+	}
+
+	if *matchFuncName != "" {
+		config.MatchFuncName = *matchFuncName
+	}
+
+	if filterflags.filter != nil {
+		config.Filter = filterflags.filter
+	}
+
+	if directiveOverride != nil {
+		config.Override = directiveOverride
+	}
+
+	if *matchFnComment != "" {
+		config.MatchFuncComment = *matchFnComment
+	}
+
+	if *sourceCodePath == "" {
+		log.Fatal("src-path can't be empty")
 	}
 
-	if *matchFuncName == "" {
-		log.Fatal("match-func can't be empty")
+	if config.DirectiveMapName == "" {
+		log.Fatal("name of directive map can't be empty")
```

Comment:
nit: use exact parameter name
```suggestion
		log.Fatal("directiveMapName can't be empty")
```